### DISCUSSION TITLE
Sync drafts between clients

### DISCUSSION THREAD

**Discussion by Lykos153** [element-hq/element-meta#502] [Ideas]
#### What would you like to do?
Start writing a draft on one client and continue it on a different client

#### Why would you like to do it?
I often start writing on mobile and then, when I realize the message is going to be longer, I switch to desktop because there it's easier to type and edit.

#### How would you like to achieve it?
By storing the (encrypted) draft on the server, maybe upon closing.

### Have you considered any alternatives?

1. Send the unfinished message and complete or edit it from a different client. => Maybe I don't want the receiver to read it in unfinished state
2. Send the unfinished message to myself. => 2x Copy, Switch room and then paste is cumbersome
3. Finish this one message on the same device. Continue the conversation on the more convenient device
4. Re-type everything from memory / copy it by hand

### Additional context

Definitely not a showstopper, but a feature that'd be really nice to have

**Comment by t3chguy** [comment]
This won't work very well until all clients use a common markdown/input dialect, Web uses Commonmark, Android uses plaintext by default but can be switched into GFM. Using different composition dialects entirely changes the representation of the message

**Comment by t3chguy** [comment]
Don't think this needs spec changes, can/should probably just be an `io.element` room-account data thing

**Comment by SISheogorath** [comment]
In order to improve privacy, would it be possible to implement this as encrypted event using direct client-to-client messages (like the ones used for key exchange) regardless of the room encryption state?

**Comment by Mguiteria** [comment]
Yes
